<?xml version="1.0" ?>

<launch>
    <!-- This allows the user to set 2 arguments when launching from the command line -->
    <arg name="robot_ip" default="172.16.0.2"/>
    <arg name="load_gripper" default="true"/>
    
    <!-- This is the hardware abstraction layer for hardware-level Panda controllers (I'm guessing) -->
    <include file="$(find franka_control)/launch/franka_control.launch">
        <!-- The HAL franka_control needs these 2 values, so we are sending our previous launch 
             arguments to the franka_control HAL. -->
        <arg name="robot_ip" value="$(arg robot_ip)"/>
        <arg name="load_gripper" value="$(arg load_gripper)"/>
    </include>

    <rosparam command="load" file="$(find hafidh_controller)/config/hafidh_controller.yaml"/>
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="frank_controller"/>
    <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find hafidh_controller)/launch/franka_description_with_marker.rviz"/>
    <node name="interactive_marker" pkg="hafidh_controller" type="interactive_marker.py" required="true" output="screen">
        <param name="link_name" value="panda_link0"/>
    </node>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false"/>
</launch>

